### PR TITLE
CPDP-1477: ensure boundedness of stream when downloading S3 files

### PR DESCRIPTION
**Pull request by mikail-khan**
@AndyHWChung as discussed. Need to investigate the guice dependency injection issue with tests.

### CODE REVIEW

File: app/uk/gov/hmrc/cgtpropertydisposals/connectors/UpscanConnector.scala [hunk 1]
```diff
@@ -55,34 +40,57 @@ trait UpscanConnector {
 }
 
 @Singleton
-class UpscanConnectorImpl @Inject() (playHttpClient: PlayHttpClient, config: ServicesConfig)(
+class UpscanConnectorImpl @Inject() (
+  playHttpClient: PlayHttpClient,
+  config: ServicesConfig,
+  implicit val mat: ActorMaterializer
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/cgtpropertydisposals/connectors/UpscanConnector.scala:46]: not sure if this is going to affect anything, but what happens if you stick this in the same parameter group as the execution context?

Comment (mikail-khan) [reply app/uk/gov/hmrc/cgtpropertydisposals/connectors/UpscanConnector.scala:46]: I did try that as well and I still get the same issue :upside_down_face:

Comment (AndyHWChung) [reply app/uk/gov/hmrc/cgtpropertydisposals/connectors/UpscanConnector.scala:46]: did you try injecting in `system: ActorSystem`?

Comment (mikail-khan) [reply app/uk/gov/hmrc/cgtpropertydisposals/connectors/UpscanConnector.scala:46]: tried that as well

**Pull request merged**